### PR TITLE
Build ARM/ARM64 tests

### PR DESCRIPTION
**Pull request by cbezault**
Enable the building of tests on `ARM`/`ARM64` using the MSVC x64_arm64 toolchain.

We also disable the `clang-cl` tests when targeting `ARM`/`ARM64` as the pre-built `llvm` binaries do not have the appropriate backend built into them.

### CODE REVIEW

File: tests/utils/stl/test/tests.py [hunk 1]
```diff
@@ -36,11 +37,14 @@ def __init__(self, suite, path_in_suite, lit_config, test_config,
 
         for flag in chain(self.cxx.flags, self.cxx.compile_flags):
             if flag[1:] == 'clr:pure':
-                self.requires.append('clr_pure') # TRANSITION, GH-798
+                self.requires.append('clr_pure')  # TRANSITION, GH-798
```
[the remark this answers is not in the document]
Comment (cbezault) [reply tests/utils/stl/test/tests.py:40]: The default `flake8` (it's like clang-format) configuration wants two spaces between code and inline comments. Maybe I should add a flake configuration file.

[the remark this answers is not in the document]
Comment (BillyONeal) [reply tests/utils/stl/test/tests.py:40]: No need then, I just wanted to understand why.

**Comment by cbezault** [comment]
Closing until we have an ARM64 device we can actually run tests on.

**Comment by AlexGuteniev** [comment]
Are virtual machines that do hardware emulation not an option?

**Comment by BillyONeal** [comment]
There is no such thing that runs arm64 windows.

**Pull request closed without merging**